src/transformation/visitors/helpers/multi.ts

```diff
@@ -0,0 +1,170 @@
+}
+
+export function transformMultiHelperReturnStatement(
+    context: TransformationContext,
+    statement: ts.ReturnStatement
+): lua.Statement | undefined {
+    if (!statement.expression) {
+        return undefined;
+    }
+
+    if (!ts.isCallExpression(statement.expression)) {
+        return undefined;
+    }
+
+    if (!isMultiHelperCallSignature(context, statement.expression)) {
+        return undefined;
+    }
+
+    const expressions = transformArguments(context, statement.expression.arguments);
+    return lua.createReturnStatement(expressions, statement);
+}
+
+function transformMultiHelperCallArguments(
+    context: TransformationContext,
+    expression: ts.CallExpression
+): lua.Expression[] | lua.Expression {
+    return isMultiHelperCallSignature(context, expression)
+        ? expression.arguments.length > 0
+            ? expression.arguments.map(e => context.transformExpression(e))
+            : lua.createNilLiteral(expression)
+        : context.transformExpression(expression);
+}
+
+export function transformMultiHelperVariableDeclaration(
+    context: TransformationContext,
+    declaration: ts.VariableDeclaration
+): lua.Statement[] | undefined {
+    if (!declaration.initializer) {
+        return undefined;
+    }
+
+    if (!ts.isCallExpression(declaration.initializer)) {
+        return undefined;
+    }
+
+    if (!isMultiReturningCallExpression(context, declaration.initializer)) {
+        return undefined;
+    }
+
+    if (!ts.isArrayBindingPattern(declaration.name)) {
+        throw InvalidMultiHelperFunctionUse(declaration.name);
+    }
+
+    if (declaration.name.elements.length < 1) {
+        throw InvalidMultiHelperFunctionUse(declaration.name);
+    }
+
+    const leftIdentifiers = declaration.name.elements.map(element => {
+        if (ts.isBindingElement(element) && ts.isIdentifier(element.name)) {
+            return transformIdentifier(context, element.name);
+        }
+
+        if (ts.isOmittedExpression(element)) {
+            return lua.createAnonymousIdentifier(element);
+        }
+
+        throw UnsupportedKind("Array Destructure Assignment Element", element.kind, element);
+    });
+    const rightExpressions = transformMultiHelperCallArguments(context, declaration.initializer);
+    return createLocalOrExportedOrGlobalDeclaration(context, leftIdentifiers, rightExpressions, declaration);
+}
+
+export function transformMultiHelperDestructuringAssignmentStatement(
+    context: TransformationContext,
+    statement: ts.ExpressionStatement
+): lua.Statement[] | undefined {
+    if (!ts.isBinaryExpression(statement.expression)) {
+        return undefined;
+    }
+
+    if (statement.expression.operatorToken.kind !== ts.SyntaxKind.EqualsToken) {
+        return undefined;
+    }
+
+    if (!ts.isCallExpression(statement.expression.right)) {
+        return undefined;
+    }
+
+    if (!isMultiReturningCallExpression(context, statement.expression.right)) {
+        return undefined;
+    }
```

Comment:
IMO single-statement condition is more readable in these cases
```suggestion
    if (!ts.isBinaryExpression(statement.expression)) return;
    if (statement.expression.operatorToken.kind !== ts.SyntaxKind.EqualsToken) return;
    if (!ts.isCallExpression(statement.expression.right)) return;
    if (!isMultiReturningCallExpression(context, statement.expression.right)) return;
```